company-lsp.el

```diff
@@ -251,6 +251,20 @@ Return a plist of (:incomplete :candidates) if cache for PREFIX
 exists. Otherwise return nil."
   (cdr (assoc prefix company-lsp--completion-cache)))
 
+(defun company-lsp--documentation (candidate)
+  "Get the documentation from the item in the CANDIDATE.
+
+The documentation can be either string or MarkupContent. This method
+will return markdown string if it is MarkupContent, original string
+otherwise. If the documentation is not present, it will return nil
+which company can handle."
+  (let* ((item (company-lsp--candidate-item candidate))
```

Comment:
nit: resolve the candidate first. It's unclear if other language server require resolution to get documentation string.

```elisp
(let* ((resolved-candidate (company-lsp--resolve-candidate candidate "documentation"))
       (item (company-lsp--candidate-item resolved-candidate))
```

Reply:
Done.